### PR TITLE
MLB Sports Scores

### PR DESCRIPTION
**Pull request by sdougbrown**
Top Secret :wink: 

Requires an internal PR.

to @bsstoner 

------

IA Page: https://duck.co/ia/view/sports_mlb_games

### CODE REVIEW

File: share/spice/sports/mlb/games/sports_mlb_games.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        if (!apiResult || !apiResult.data || !apiResult.data.games || !apiResult.data.games.length) {
+            return Spice.failed('mlb_games');
+        }
+        
+        DDG.require('moment.js', function(){
+            Spice.add({
+                id: 'mlb_games',
+                name: 'MLB Games',
+
+                allowMultipleCalls: true,
```
Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Is this right? The api doesn't return a next param in the response so there's no load more tile currently showing.

Comment (sdougbrown) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: We're planning on adding that later but yeah we don't need it now.

File: share/spice/sports/mlb/games/sports_mlb_games.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                        return l("Tomorrow");
+                        break;
+                    case 1:
+                        return l("Yesterday");
+                        break;
+                    default:
+                        return false;
+                        break;
+                }
+            }
```
Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: This was kind of hard to follow. Maybe using the moment functions would be easier to read?

```
var tomorrow = today.add(1, 'd'),
    yesterday = today.sub(1, 'd');

if (date.sameAs(today, 'd')) { 
    return 'Today';
} else if (date.sameAs(yesterday, 'd') {
    return 'Yesterday';
} else if (date.sameAs(tomorrow, 'd') {
    return 'Tomorrow';
}
```

File: share/spice/sports/mlb/games/sports_mlb_games.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                data: gameData(apiResult.data),
+
+                from: apiResult.from,
+                signal: apiResult.signal,
+
+                meta: {
+                    idField: 'id',
+                    showMoreAt: true,
+                    sourceIcon: false,
+                    sourceUrl: "http://www.bleacherreport.com/",
```
Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: It would be nice if we could link to the team page for the team they searched for, but I don't see an easy way to do it given the current api response. Maybe we should at least link to the league page? i.e. `http://www.bleacherreport.com/mlb`?

File: share/spice/sports/mlb/games/sports_mlb_games.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                    i = 0;
+                // for historical games on the first call, 
+                // only add the most relevant game and one previous
+                // (if available)
+                for (; i < data.games.length; i++) {
+                    if (data.games[i].id === data.most_relevant_game_id) {
+                        // most relevant first
+                        games.push(data.games[i]);
+                        // one historical game if the layout allows for it
+                        if (!DDG.device.isMobileLandscape() && data.games[i-1]) { games.push(data.games[i-1]); }
```
Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Why do we show an extra game in portrait but not landscape? Add comments maybe?

File: share/spice/sports/mlb/games/sports_mlb_games.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                signal: apiResult.signal,
+
+                meta: {
+                    idField: 'id',
+                    showMoreAt: true,
+                    sourceIcon: false,
+                    sourceUrl: "http://www.bleacherreport.com/",
+                    sourceName: 'Bleacher Report',
+                    secondaryText: '<span class="tx-clr--grey-dark">Data from SportsData</span>',
+                    hideModeSwitch: true,
```
Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Why do we hide it completely here but just disable everywhere else (i.e. weather)? Seems like inconsistent UI.

File: share/spice/sports/mlb/games/sports_mlb_games.js (hunk 3 above)
Comment (sdougbrown) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: yeah that's probably a better idea

File: share/spice/sports/mlb/games/sports_mlb_games.js (hunk 5 above)
Comment (sdougbrown) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: We just added the ability to hide it so nothing else is up to speed yet?

Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Cool, just double checking that we intend to start hiding that thing going forward. We had initially hid it in bttf and then added it back, disabled.

Comment (sdougbrown) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Yeah I think going forward if an IA is never going to allow expand then it should be hidden - it should only be disabled if there aren't enough items for grid mode.

File: share/spice/sports/mlb/games/sports_mlb_games.js (hunk 2 above)
Comment (sdougbrown) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Easier to read, but totally broken:
moment-days-suck

Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Really? The comment doesn't make sense to me. You want rounded down dates.

Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Try this, it seems to work for me:
```
var today = moment(),
    tomorrow = moment().add(1, 'd'),
    yesterday = moment().subtract(1, 'd');

if (date.sameAs(today, 'd')) { 
    return 'Today';
} else if (date.sameAs(yesterday, 'd') {
    return 'Yesterday';
} else if (date.sameAs(tomorrow, 'd') {
    return 'Tomorrow';
}
```

Comment (sdougbrown) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: `sameAs` wasn't working for me at all when I tested it for that screenshot, I had to use `isSame`...

File: share/spice/sports/mlb/games/sports_mlb_games.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+                from: apiResult.from,
+                signal: apiResult.signal,
+
+                meta: {
+                    idField: 'id',
+                    showMoreAt: true,
+                    sourceIcon: false,
+                    sourceUrl: "http://www.bleacherreport.com/mlb",
+                    sourceName: 'Bleacher Report',
+                    secondaryText: '<span class="tx-clr--grey-dark">Data from SportsData</span>',
```
Comment (moollaza) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: Should be use the fancy, new `moreText` here?

Comment (sdougbrown) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: `moreText` is for a detail area (e.g. wiki), `secondaryText` is for the metabar.  (There are often cases where these would need to be different - e.g. forecast temperature change).

File: share/spice/sports/mlb/games/sports_mlb_games.js (hunk 2 above)
Comment (bsstoner) [inline share/spice/sports/mlb/games/sports_mlb_games.js]: yea, my bad, should be `isSame`, was typing from memory. Did it work though? Can we make the change?

**Comment by daxtheduck** [comment]
## [MLB Games](https://duck.co/ia/view/sports_mlb_games)

**Description**: Boxscores for major league baseball games, provided by Sportsradar

**Example Query**: [](https://beta.duckduckgo.com/?q=)

**Tab Name**: MLB

**Source**:  

*These are the important fields from the IA page.  Please check these for errors or missing information and update the [IA page](https://duck.co/ia/view/sports_mlb_games)*

---
*This is an automated message which will be updated as changes are made to the [IA page](https://duck.co/ia/view/sports_mlb_games)*
